packages/ethereum/contracts/utils/ECDSA.sol

```diff
@@ -1,97 +1,18 @@
- * of the private keys of a given address.
  */
 library ECDSA {
-    // y^2 = x^3 + 7 mod p, where p is FIELD_ORDER
-    uint256 constant FIELD_ORDER = 0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f;
     uint256 constant CURVE_ORDER = 0xfffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364141;
     uint256 constant HALF_CURVE_ORDER = (CURVE_ORDER - 1) / 2;
 
     /**
-     * @dev Computes the Ethereum address from a public key given as an
-     * uncompressed EC-point.
-     */
-    function pubKeyToEthereumAddress(uint256 x, uint256 y) internal pure returns (address) {
-        require(validate(x, y), "Point must be on the curve.");
-        return address(bytes20(bytes32(keccak256(abi.encodePacked(x, y)) << 96)));
-    }
-
-    /**
-     * @dev Computes the Ethereum address from a public key given as a
-     * compressed EC-point.
-     */
-    function compressedPubKeyToEthereumAddress(uint256 compressedX, uint8 odd) internal returns (address) {
-        (uint256 x, uint256 y) = decompress(compressedX, odd);
-        return pubKeyToEthereumAddress(x, y);
-    }
-
-    function compress(uint256 x, uint256 y) internal pure returns (uint256, uint8) {
-        return (x, uint8(y % 2));
-    }
-
-    /**
-     * @dev Decompresses a compressed elliptic curve point and
-     * returns the uncompressed version.
-     * @notice secp256k1: y^2 = x^3 + 7 (mod p)
-     * "Converts from (x, 1 / 0) to (x,y)"
+     * @dev Computes the Ethereum address from an uncomporessed public key.
      */
-    function decompress(uint256 x, uint8 odd) internal returns (uint256, uint256) {
-        uint256 sqrY = addmod(7, mulmod(mulmod(x, x, FIELD_ORDER), x, FIELD_ORDER), FIELD_ORDER);
-
-        uint256 sqrtExponent = (FIELD_ORDER + 1) / 4;
-
-        uint256 y;
-
-        /* solhint-disable no-inline-assembly */
-        assembly {
-            // free memory pointer
-            let memPtr := mload(0x40)
-
-            // length of base, exponent, modulus
-            mstore(memPtr, 0x20)
-            mstore(add(memPtr, 0x20), 0x20)
-            mstore(add(memPtr, 0x40), 0x20)
-
-            // assign base, exponent, modulus
-            mstore(add(memPtr, 0x60), sqrY)
-            mstore(add(memPtr, 0x80), sqrtExponent)
-            mstore(add(memPtr, 0xa0), FIELD_ORDER)
-
-            // call the precompiled contract BigModExp (0x05)
-            let success := call(gas(), 0x05, 0x0, memPtr, 0xc0, memPtr, 0x20)
-
-            switch success
-                case 0 {
-                    revert(0x0, 0x0)
-                }
-                default {
-                    y := mload(memPtr)
-                }
-        }
-        /* solhint-enable no-inline-assembly */
-
-        bool isOdd = y % 2 == 1;
-
-        if ((isOdd && odd == 0) || (!isOdd && odd == 1)) {
-            y = FIELD_ORDER - y;
-        }
-
-        return (x, y);
-    }
-
-    function validate(uint256 x, uint256 y) public pure returns (bool) {
```

Comment:
Aren't we validating pubkeys anymore? This need to change

Reply:
https://github.com/hoprnet/hoprnet/blob/d1f07b8a5ad6222ceeba44fc7925e1bf11253c71/packages/ethereum/contracts/HoprChannels.sol#L400
please review the PR thoroughly and provide constructive feedback

Reply:
This is constructive feedback because you change the public key check from checking a EC point to checking an arbitrary bitstring. As I wrote, a simple length check can be sufficient if you don't want to check EC math